packages/e2e/lib/e2e_driver.dart

```diff
@@ -4,15 +4,59 @@ import 'dart:io';
 import 'package:e2e/common.dart' as e2e;
 import 'package:flutter_driver/flutter_driver.dart';
 
-Future<void> main() async {
+Future<void> main() async => e2eDriver();
+
+/// Adaptor to run E2E test using `flutter drive`.
+///
+/// `timeout` controls the longest time waited before the test ends.
+/// It is not necessarily the execution time for the test app.
+/// The test can and usually runs for shorter time if the test ends early.
+///
+/// `traceTimeline` flag controls if timeline and timeline summary should be
+/// collected.
+///
+/// `testName` is used as the file name for the output timeline files.
+///
+/// To an E2E test `<test_name>.dart` using `flutter drive`, put a file named
+/// `<test_name>_test.dart` in the app's `test_driver` directory:
+///
+/// ```dart
+/// import 'dart:async';
+///
+/// import 'package:e2e/e2e_driver.dart' as e2e;
+///
+/// Future<void> main() async => e2e.e2eDriver();
+///
+/// ```
+Future<void> e2eDriver({
+  Duration timeout = const Duration(minutes: 1),
```

Comment:
Generally we try to avoid timeouts in tests, because they're a source of flakiness.

That said, I'm not sure that 1 minute is long enough here - integration tests can easily take more (for example, they may have to wait on network requests, animations to finish, etc). In the devicelab we use a default of 15 minutes for the entire test run. This should probably be higher by default, maybe something like an hour.

We should also document that these parameters must not be null.

Reply:
This 1min is what used to be hard coded in `main`. Any suggestion what value I should put here?

Reply:
And for the documents I added an `assert(timeout != null);`. I'm expecting it will be obvious since the implementation will be shown together with detailed documents anyway.